core/backend/src/IModelExporter.ts

```diff
@@ -309,11 +310,14 @@ export class IModelExporter {
           readyToExport = schema.fullName === BisCoreSchema.schemaName; // schemas prior to BisCore are considered *system* schemas
         }
         if (readyToExport && this.handler.callProtected.shouldExportSchema(schema)) {
-          Logger.logTrace(loggerCategory, `exportSchema(${schemaName})`);
-          this.handler.callProtected.onExportSchema(schema);
+          schemasToExport.push(schema);
         }
       }
     });
+    await Promise.all(schemasToExport.map(async (schema) => {
+      Logger.logTrace(loggerCategory, `exportSchema(${schema.fullName})`);
+      return this.handler.callProtected.onExportSchema(schema);
+    }));

```

Comment:
Because of dependencies, the order that schemas are exported to the target must match the order that they were imported into the source. Is the above Promise.all code guaranteed to maintain that order?

Reply:
good point, it is not guaranteed. A more complex barrier could do it and keep the parallelization, but I think I'll just remove the Promise.all.

Reply:
Were you going to refactor to not use Promise.all?

Reply:
it does run `onExportSchema` using `Promise.all` here, was there somewhere else you were referring to?

Reply:
Schema order is important and I thought Promise.all was not guaranteed to have a deterministic order? Instead you could iterate the `schemasToExport` array and await each call to `onExportSchema`.

Reply:
I thought schema order was only important for the actual import, and the import should still happens in the same order as it did before this PR even with the `Promise.all`.
Though now that I think about it, it only happens in whatever order the implementation of `readdirSync` uses which is platform dependent. Afaict, the previous implementation of `processSchemas` did not order the imports differently.
Am I missing something?

Reply:
If the `meta.ECSchemaDef` query is in schema topological/reference order then that's probably it. Yes this may export them out of order. I suppose I will just remove any parallelism for the time being.

Reply:
I also now realize you wrote "refactor to *not* use Promise.all", I somehow missed the "not" (even though the comment that way is contextually pointless). Either way fixed.

Reply:
Sorry, you can probably keep things as you have them. Schema order is important if you import them one by one. Now that you are importing them as a collection after all exports, then this should be okay since importSchemas doesn't care about the order - it just has to have dependencies in the list.

Reply:
I did forget that passing them as a list lets the native db handle sorting them before import. Alright well in that case I'll revert my revert etc and keep exporting schemas parallel